src/PlayingState.cpp

```diff
@@ -272,6 +268,27 @@ void PlayingState::Listen() {
     int note_number = ev.NoteNumber();
     string note_name = MidiEvent::NoteName(note_number);
 
+    if (ev.NoteVelocity() != 0)
+      m_last_notes.push_back(NoteWithTime{static_cast<int>(ev.NoteNumber()), ev.NoteTimestamp()});
+
+    if (m_last_notes.size() == 2)
+    {
+      const std::set<int> ref_notes = {m_keyboard->GetMinPlayableNote(),m_keyboard->GetMaxPlayableNote()};
+      const std::set<int> last_notes = {m_last_notes[0].note, m_last_notes[1].note};
+      const unsigned long max_ts = std::max(m_last_notes[0].timestamp, m_last_notes[1].timestamp);
+      const unsigned long min_ts = std::min(m_last_notes[0].timestamp, m_last_notes[1].timestamp);
+      using namespace std::chrono;
+      constexpr unsigned long max_deltatime = duration_cast<nanoseconds>(milliseconds{250}).count();
+      if ((last_notes == ref_notes) && ((max_ts-min_ts) < max_deltatime))
+        m_paused = !m_paused;
+
+      m_last_notes = {};
```

Comment:
this code doesn't always correctly detect that both keys were simultaneously pressed

for example, if we press any key (just once), the press both keys at the same time.. it will not pause.

i suggest instead of clearing the vector, to just erase the first note instead
```cpp
m_last_notes.erase(m_last_notes.begin());
```

Reply:
You are right, but in this case we don't even need an array just the last note.
And we can also use the `std::abs` to check if the combination is the first and last playable note.

Reply:
nice :+1: